src/HTTP/HttpConnectionHandler.cs

```diff
@@ -14,7 +14,8 @@ public class HttpConnectionHandler : IProtocolConnectionHandler {
         public void HandleRequest(IChorizoSocket socket) {
             var bytes = SocketReader.ReadSocket(socket);
             var request = DataParser.Parse(bytes);
-            var response = RequestProcessor.Retrieve(request);
+            var requestWithBody = SocketReader.ReadBody(socket, request);
```

Comment:
`ReadBody` seems like it fits better as part of `Parse` since you're already extracting out all the parts of an HTTP Request